### PR TITLE
support extension targets in deployment_target rule

### LINKED ISSUE #4004: deployment_target rule should handle iOSApplicationExtension & friends
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Besides the identifiers we already support, we should also handle `iOSApplicationExtension` and other identifiers mentioned in https://docs.swift.org/swift-book/ReferenceManual/Attributes.html

```swift
@available(iOSApplicationExtension 14.0, *)
class Foo {}
```

### PR DESCRIPTION
**Pull request by tahabebek**
Closes #4004

### CODE REVIEW

**Comment by marcelofabri** [comment]
IMO we should default them to their counterpart, e.g. if I set `ios_deployment_target: 13.0`, that should also change `iOSApplicationExtension_deployment_target` unless `iOSApplicationExtension_deployment_target` is explicitly defined

**Comment by tahabebek** [comment]
@marcelofabri, I have addressed the issue of defaulting.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@
 
 #### Enhancements
 
+* Support iOSApplicationExtension, macOSApplicationExtension, watchOSApplicationExtension, and tvOSApplicationExtension identifiers in `deployment_target` rule. To configure the rule for these identidiers, you need to use these keys: `iOSApplicationExtension_deployment_target`, `macOSApplicationExtension_deployment_target`, `watchOSApplicationExtension_deployment_target`, `tvOSApplicationExtension_deployment_target`.
```
Comment (SimplyDanny) [inline CHANGELOG.md:15]: Please add line breaks after about every 80 characters and add two additional spaces at the end of the description. They are required to render the CHANGELOG properly:
```suggestion
* Support `iOSApplicationExtension`, `macOSApplicationExtension`,
  `watchOSApplicationExtension`, and `tvOSApplicationExtension` identifiers
  in the `deployment_target` rule. To configure the rule for these identifiers,
  you need to use the keys `iOSApplicationExtension_deployment_target`,
  `macOSApplicationExtension_deployment_target`,
  `watchOSApplicationExtension_deployment_target`, and
  `tvOSApplicationExtension_deployment_target`.  
```

**Review by SimplyDanny — CHANGES_REQUESTED** [review]

**Comment by SimplyDanny** [comment]
> @marcelofabri, I have addressed the issue of defaulting.

You may also add this detail to the CHANGELOG entry.

**Comment by tahabebek** [comment]
@SimplyDanny, I have accepted your suggestions and added the details about defaulting to the changelog entry.

**Review by SimplyDanny — APPROVED** [review]

**Pull request merged**